.github/workflows/DiffTool-CI.yml

```diff
@@ -0,0 +1,145 @@
+name: diff-java-tool PR checks
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  # Allows manual triggering test changes in checkstyle/test-configs
+  workflow_dispatch:  
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    env:
+      JAVA_VERSION: '21'
+      CONFIG_DIR: '.ci-config'
+      DIFFTOOL_JAR: "diff-java-tool.jar"
+      PATCH_DIFF_TOOL_VERSION: "0.1-SNAPSHOT"
+
+    steps:
+      - name: Checkout Main Repository
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Checkout Checkstyle Repository
+        uses: actions/checkout@v4
+        with:
+          repository: checkstyle/checkstyle
+          path: checkstyle
```

Comment:
please put clones and downloads to `.ci-temp` folder to not pollute in git status of test-configs repo.

Reply:
done